Mathlib/GroupTheory/MonoidLocalization/Basic.lean

```diff
@@ -1297,6 +1295,36 @@ theorem mulEquivOfQuotient_symm_monoidOf (x) :
     (mulEquivOfQuotient f).symm (f.toMap x) = (monoidOf S).toMap x :=
   f.lift_eq (monoidOf S).map_units _
 
+/-- The localization of a finitely generated monoid at a finitely generated submonoid is
+finitely generated. -/
+@[to_additive "The localization of a finitely generated monoid at a finitely generated submonoid is
+finitely generated."]
+lemma fg [Monoid.FG M] (hS : S.FG) : Monoid.FG <| Localization S := by
+  rw [← Monoid.fg_iff_submonoid_fg] at hS
+  let antidiag : M × S →* Localization S := {
```

Comment:
Do you think it's worth making this a standalone definition, and proving its surjectivity as a lemma? I'm happy either way, but I can vaguely imagine someone using surjectivity of this map for something else...?